src/mocks/data.ts

```diff
@@ -3,6 +3,689 @@ import { DAYS_AGO } from "../test/utils";
 import { RegisteredUserDTO, IsaacEventPageDTO, UserSummaryWithGroupMembershipDTO } from "../IsaacApiTypes";
 import { EventBookingDTO } from "../IsaacApiTypes";
 
+export const mockRandomQuestions = [
```

Comment:
The type is missing.

Reply:
This is another area where I believe there are some complex typing problems in the codebase. `IsaacQuestionPageDTO[]` is the expected type here, but that type expects `children` to be `ContentDTO[]`. In reality in this instance children can be `QuestionDTO` (which includes `hints`). 

If I update it so that `children` is `(QuestionDTO | ContentDTO)[];`, then `children` in `hints` start objecting because that's typed to be `ContentBaseDTO[]`. If I change that to `ContentDTO[]` it then complains about `src` in some of the children of hints. `src` is only expected from `MediaDTO` but that would then mean making `children` to be `(QuestionDTO | ContentDTO | MediaDTO)[]` 

But theeeen it objects about `choices` which only comes from `IsaacMultiChoiceQuestionDTO`. It feels a bit like playing whack-a-mole.

Setting `children` of `ContentDTO` to this: `(QuestionDTO | ContentDTO | MediaDTO | IsaacMultiChoiceQuestionDTO)[];` appears to solve it, but that just feels wrong, particularly because i've only used multiple choice question as the test data. If I change that test data to what is returned from a different question type, I suspect it would throw up new errors. 

In summary... it works without setting the type, and unless I'm missing something obvious I think it's going to be a headache for a future ticket to fix these types.

Reply:
Thanks for the detailed explanation! I think you're right, this sounds like something that should be solved in its own ticket.

Having said that, it sounds like the only cause of errors here is the `children` property. My question is: are the children of `IsaacQuestionPageDTO` **ALWAYS** `QuestionDTO`s? If so, then an easy solution is to add the following to `IsaacQuestionPageDTO`:

```typescript
children?: QuestionDTO[];
```
That way, you'll override the type of `children` for this `IsaacQuestionPageDTO` while leaving `ContentDTO` intact. Also, if `IsaacQuestionPageDTO` is always guaranteed to return a `children` property (even if it's an empty array), then you should remove the question mark.

But if this does not work, then please create a new ticket to handle this and ignore my comments.

Reply:
Unfortunately no, the children of `IsaacQuestionPageDTO` can be a variety of things. Just taking the one sample question that I used to build the mock data, it has a `ContentDTO` and `QuestionDTO` in the array. I'm assuming that may vary between different questions. If set it as this 

`children?: (ContentDTO | QuestionDTO)[];` 

Then it still objects about the children of `hints` because apparently `ContentBaseDTO` shouldn't have children, and those children appear to be an assortment of types like `ContentDTO`, `MediaDTO` (and likely more if I looked at more data). 

I've created a ticket for it now as it feels like one that's going to take some time to resolve: https://github.com/isaaccomputerscience/isaac-cs-issues/issues/237